src/ess/nmx/mcstas_loader.py

```diff
@@ -0,0 +1,82 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from typing import Iterable, NewType, Optional
+
+import scipp as sc
+import scippnexus as snx
+
+PixelIDs = NewType("PixelIDs", sc.Variable)
+InputFilepath = NewType("InputFilepath", str)
+NMXData = NewType("NMXData", sc.DataArray)
+
+# McStas Configurations
+MaximumProbability = NewType("MaximumProbability", int)
+DefaultMaximumProbability = MaximumProbability(100_000)
+
+
+def _retrieve_event_list_name(keys: Iterable[str]) -> str:
+    prefix = "bank01_events_dat_list"
+
+    # (weight, x, y, n, pixel id, time of arrival)
+    mandatory_fields = 'p_x_y_n_id_t'
+
+    for key in keys:
+        if key.startswith(prefix) and mandatory_fields in key:
+            return key
+
+    raise ValueError("Can not find event list name.")
+
+
+def _copy_partial_var(
+    var: sc.Variable, idx: int, unit: Optional[str] = None, dtype: Optional[str] = None
+) -> sc.Variable:
+    """Retrieve a property from a variable."""
+    original_var = var['dim_1', idx].copy()
+    var = original_var.astype(dtype, copy=False) if dtype else original_var
```

Comment:
One final question. Can we use
```suggestion
    var = var['dim_1', idx].astype(dtype or var.dtype , copy=True)
```
to avoid up to one copy?